addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -595,7 +596,7 @@ export class HtmlField extends Component {
         }
     }
     async _onReadonlyClickChecklist(ev) {
-        if (ev.offsetX > 0) {
+        if (!isMouseInsideCheckbox || this._isDirty()) {
```

Comment:
You're just checking that the function `isMouseInsideCheckbox` was properly imported?

Reply:
I have updated it!

Reply:
It worries me that it passed testing (automated and human) while the first condition was always false. Could we add a test for this? I think another run of Testing IN/BE would be good after the changes.

Reply:
I have considered creating a QUnit test case for this, but I am unable to execute the final step necessary for this task. This step involves navigating back from the current record and reopening it to verify if the checklist has been saved correctly.

Reply:
I attempted to create a tour for this but encountered an issue. The problem occurs because the `tooltip` added during the tour process prevents the code responsible for saving the checklist from executing.
I've added a condition where if `_isDirty()` returns true, the code returns from the `_onReadonlyClickChecklist` function before the code responsible for saving the checklist is executed. and because of the `tooltip`, `_isDirty()` returns true.

Reply:
Maybe you can push the commit with the failing tour so I can take a look, because like this it's hard for me to tell what's to be done.

Reply:
I have pushed the tour commit. You can take a look at it

Reply:
Your commit is supposed to be about checking a checkbox _in readonly mode_. Why is your tour testing _in edit mode_? If you were in readonly mode you shouldn't need worry about `isDirty`. Which begs the question: why did you add these `isDirty` conditions in the first place?

Reply:
I had a discussion with @dmo-odoo in the task _lognote_ regarding _readonly_ and go to know that **from 16.0**, the `html_field` is only in readonly mode if it lacks edit rights. There's no scenario where it possesses edit rights but it can goes into readonly mode also. In our case, since we can edit in notes, I don't believe it can switch to readonly mode.

That is why i have use `_isDirty()` here. with this ,  I am determining that if it returns false, the page is in readonly mode because the user hasn't made any edits in the record. However, if it returns true, it indicates that the user has edited the record, so it can be considered as being in edit mode.

> Your commit is supposed to be about checking a checkbox in readonly mode. Why is your tour testing in edit mode? 

I don't think i am checking it in edit mode or i am missing something, because i have created a step to save the record before the checkbox is checked so to make sure that when checkbox is checked at that time it is not in edit mode.
or because of trigger it goes back to in edit mode?

Reply:
I think there's a misunderstanding here about _readonly_ and `_isDirty`. If `_isDirty` is true, that would indeed probably indicate that you're in edit mode because you're not supposed to be able to make changes in readonly, by definition. But the converse is not true! `_isDirty` being false **does not mean the record is _readonly_**: it just means that no change has been made since the last time the record was saved.

So you are in fact testing in edit mode.

Anyway, now having read again your discussion with @dmo-odoo in the task, and the steps to reproduce, I understand it's not really a question of readonly/edit but probably rather that clicking the checkbox doesn't focus the editor so that clicking on the breadcrumb after that doesn't cause a blur event, and nothing is saved. That or the check mark doesn't cause the record to register as dirty, but that would be strange. Can you check if the first hypothesis is true ? I'll take a look at your approach based on your answer. Thanks.

(Either way, to reproduce the steps, you need to leave the page before clicking on the checkbox.)

Reply:
Yes. In this situation, the _blur_ event isn't being triggered, and it's because the `mousedown` event on the checkbox fails to focus the editor. This issue is likely caused by [this](https://github.com/odoo/odoo/blob/82ac6aee2702eb3d4ba3b01047e9cd3a3d66bfa5/addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js#L4348[this]). because after removing it, focus is properly set. As a result, both the _blur_ event is triggered when the back button is clicked, and the checklist is also successfully saved.

Reply:
Any idea why that line had been added? It seems like the right fix is indeed to remove it.

Reply:
I don't know why it was added, but it has been there since this [PR](https://github.com/odoo-dev/odoo-editor/commit/721055e3afca2d5966069657634ce6a1381aed9b)

Reply:
The tour is still not functioning properly. Upon investigation, it appears that dispatching `mousedown` and `click` events within the tour are not triggering the `focus` and `blur` events as expected.

Reply:
That might be, you may have to trigger them yourself.

Reply:
Yes, it might work, but manually triggering those events isn't a good idea. If I were to do so, the tour would still proceed even if I were to remove my fix. This is because what I've done is removed the `ev.preventDefault` inside `mousedown`, which is responsible for preventing focus on the checklist when the checkbox is clicked.

Reply:
@dmo-odoo Does this ring a bell? @dhba-odoo is writing a tour that relies on dispatching `mousedown` and `click` events and they don't seem to impact the focus (or at least cause `focus` and `blur` events). Any chance you'd know of a workaround?

Reply:
Maybe because the events are not trusted, like the arrow keys? What about calling focus manually just for the purpose of the test? Before doing that, check for other tests that might do the same, as I'd be surprised that this would be the first time.

Reply:
@dmo-odoo

> What about calling focus manually just for the purpose of the test?

That would kind of defeat the purpose of the test because the fix he's testing involves not preventing a focus event. But agreed on the other part (checking other tests) @dhba-odoo

Reply:
@dmo-odoo @Zynton I've looked into the other test cases, and most of the time the mouseup event is triggered after the mousedown event, or manual focus and blur events are used. I have tried both approaches and only manually triggering focus and blur events works. However, this approach wouldn't work for our specific need because triggering focus and blur manually would still pass the test case without my fix too.

Reply:
@dhba-odoo What happens if you try to reproduce the same case in a jsfiddle and manually trigger a mousedown or click in js? Do you get focus/blur events then ? If yes, then something in our code is preventing those events from firing and we need to find what. If no, then we can't test this bug due to that limitation. cc @Zynton

Reply:
I checked on `JSFiddle`, and the same thing happens there too. When manually triggering a `mousedown` event, it correctly calls the focus event. However, when using dispatchEvent, only the `mousedown` event is triggered, and the focus event doesn't occur. The same problem occurs with the `click` and `blur` events as well.

Reply:
I don't, I was going to say the same thing as you.